### ISSUE TITLE
signet mining pool identifiers

### ISSUE THREAD

**Issue by ajtowns** [mempool/mempool#2903]
Hi!

It would be neat if we could identify signet blocks by mining pool. One complication is it that it would be especially neat if this worked equally well for custom signets (eg https://plebnet.wiki/wiki/Plebnet_Playground). So two options come to mind.

Ideally:

 * signet miners pick a random identifier, eg "--poolid=aj" and tag their blocks correspondingly, with "/signet:aj/" in the coinbase
 * mempool.space picks up these entries from the coinbase by matching on "/signet:([A-Za-z0-9_-]+)/" and creates pool entries for them

Does something like that seem feasible?

The other alternative seems to me to be:

 * we hardcode 8 signet pools into pools.json, "Signet Pool 1" through "Signet Pool 8"
 * signet miners pick a pool number, and tag their blocks with "/signet:3/" etc
 * the code just works as-is

I've been tagging blocks with [signet:2](https://mempool.space/signet/block/0000007885facb0d632f24cc7dbba4791a81d61784dbd6a97a07b484d3717326) and [signet:3](https://mempool.space/signet/block/000000df2d50f49348e1bdd05a6808aec43536d4f9524c2d0c084bbd01ee5597) recently, if you want to test.

**Comment by wiz** [comment]
Yeah cool idea, I suppose it would be easy to just add signet: tags into the main pools.json file

**Comment by 0xB10C** [comment]
@nymkappa and I have been working on combining the efforts of maintaining a dataset of known pools in  https://github.com/mempool/mining-pools/issues/25. He's also rewriting the mining pool data parser  https://github.com/mempool/mempool/pull/2869. 

If mempool-space ends maintaining and syncing a fork of "my" [known-pools](https://github.com/0xB10C/known-mining-pools) repo, as mentioned as an option in https://github.com/mempool/mempool/pull/2869#issuecomment-1369714397, it could just be a new JSON file per SigNet pool mapping signet:n to e.g.  "signet inquisition pool" and "signet core pool" and "signet kalle pool". I'd also be happy to include these mappings in "my" data, but would probably want to split them up from the mainnet data in some way.

**Comment by nymkappa** [comment]
I also prefer having a different json file so we can keep the mainnet pools.json clean.

On mempool end we can point to any pools.json file we want so we will change that config value accordingly for signet.

**Comment by ajtowns** [comment]
Oh, I suppose anyone pointing mempool.space code at a custom signet could just also do a custom pools.json anyway. That does seem pretty simple.

**Comment by ajtowns** [comment]
Would it make sense to PR a `pools-signet.json` file to the mining-pools repo, then y'all can just update the signet config to point at that? (I'm a bit confused as to whether that's the right repo / file, or if there's a newer new thing or something)

**Comment by nymkappa** [comment]
> Would it make sense to PR a `pools-signet.json` file to the mining-pools repo, then y'all can just update the signet config to point at that? (I'm a bit confused as to whether that's the right repo / file, or if there's a newer new thing or something)

I believe https://github.com/bitcoin-data/mining-pools is more appropriate for that purpose now. We tabled that testnet/signet mining pool tagging feature on our end for now.

**Comment by 0xB10C** [comment]
Happy to move the current mainnet pool definitions to `pools/mainnet` and create `pools/signet` in https://github.com/bitcoin-data/mining-pools if you want @ajtowns.

**Comment by ajtowns** [comment]
I opened a PR https://github.com/bitcoin-data/mining-pools/pull/82 that should allow you to get a json file by running:

```sh
$ python3 contrib/generate-json-pool-list.py pools-list-signet.json signet-pools/
Using pools-list-signet.json as output file name.
```

Does that result in something mempool.space can use?

**Comment by 0xB10C** [comment]
There are now these JSON files that the signet mempool.space can use

https://github.com/bitcoin-data/mining-pools/blob/generated/signet-pools.json
https://github.com/bitcoin-data/mining-pools/blob/generated/signet-pool-list.json
